plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/ArtifactRetriever.java

```diff
@@ -146,30 +120,32 @@ public ArtifactRetriever() {
   }
 
   /**
-   * Returns the latest published artifact version, or null if there is no such version.
+   * Returns the latest published non-beta artifact version, or null if there is no such version.
    */
   public ArtifactVersion getLatestArtifactVersion(String groupId, String artifactId) {
     return getLatestIncrementalVersion(idToKey(groupId, artifactId), null);
   }
 
   /**
-   * Returns the latest published artifact version in the version range, or null if there is no such
-   * version.
+   * Returns the latest published non-beta artifact version in the version range, 
+   * or null if there is no such version.
```

Comment:
> I don't see it either. I thought it was another potentially useful util method.

Sorry, that was a comment at a wrong place.

If you look at `getLatestIncrementalVersion()`, it first tries to get a latest release (without a qualifier) version:

```java
      NavigableSet<ArtifactVersion> allVersions = availableVersions.get(coordinates);
      ArtifactVersion latest = getLatestReleasedVersion(allVersions);
      if (range == null || range.containsVersion(latest)) {
        return latest;
      }
```

If that fails, it falls back to include qualifier versions. 

```java
      NavigableSet<ArtifactVersion> allVersions = availableVersions.get(coordinates);
      ArtifactVersion latest = getLatestInRange(range, allVersions);
      return latest;
```